test/unit/foreman/templates/snippets/container_certs_setup_test.rb

```diff
@@ -0,0 +1,43 @@
+require 'test_helper'
+
+class ContainerCertsSetupTest < ActiveSupport::TestCase
+  def renderer
+    @renderer ||= Foreman::Renderer::SafeModeRenderer
+  end
+
+  def render_template(registration_host)
+    @snippet ||= File.read(Rails.root.join('app', 'views', 'unattended', 'provisioning_templates', 'snippet', 'container_certs_setup.erb'))
+
+    source = OpenStruct.new(
+      name: 'Test',
+      content: @snippet
+    )
+
+    scope = Class.new(Foreman::Renderer::Scope::Provisioning).send(
+      :new,
+      host: nil,
+      source: source,
+      variables: {
+       registration_host: registration_host,
+      })
+
+    renderer.render(source, scope)
+  end
+
+  test 'should render successfully with valid registration_host' do
+    registration_host = 'registry.example.com'
+    output = render_template(registration_host)
+    assert_match(/hostname="registry.example.com"/, output)
+    assert_match(/Create directory for certificates\ncert_dir="\/etc\/containers\/certs\.d\/\$hostname"/, output)
+    assert_match(%r{ln -sf "\$ca_cert" "\$cert_dir/ca-bundle\.crt"}, output)
+    assert_match(%r{ln -sf "\$key_file" "\$cert_dir/client\.key"}, output)
+    assert_match(%r{ln -sf "\$cert_file" "\$cert_dir/client\.cert"}, output)
+    assert_match(/Container certificate setup completed successfully\./, output)
```

Comment:
Instead of regexes you can also use `assert_includes` with plain strings. Saves you a lot of escaping :).